tests/telemetry/test_telemetry.py

```diff
@@ -45,23 +47,43 @@ def setup_telemetry_forpyclient(duthost):
     else:
         logger.info('client auth is false. No need to restart telemetry')
 
-def verify_telemetry_dockerimage(duthost):
+@pytest.fixture(scope="module", autouse=True)
+def verify_telemetry_dockerimage(duthosts, rand_one_dut_hostname):
     """If telemetry docker is available in image then return true
     """
     docker_out_list = []
+    duthost = duthosts[rand_one_dut_hostname]
     docker_out = duthost.shell('docker images docker-sonic-telemetry', module_ignore_errors=False)['stdout_lines']
     docker_out_list = get_list_stdout(docker_out)
     matching = [s for s in docker_out_list if "docker-sonic-telemetry" in s]
-    return (len(matching) > 0)
+    if not (len(matching) > 0):
+        pytest.skip("docker-sonic-telemetry is not part of the image")
+
+@pytest.fixture
+def post_setup_streaming_telemetry(duthosts, rand_one_dut_hostname, localhost,  ptfhost):
```

Comment:
Why name it 'post' setup? It is a little bit confusing to me because it is executed before each tests, right?

Reply:
Good catch! I think `setup_streaming_telemetry(...)` is a better one.